src/tls/acme/dns/cloudflare.rs

```diff
@@ -140,14 +141,19 @@ impl DnsManager for Cloudflare {
             .await
             .context("unable to find records")?;
 
-        // Delete all matching records
+        // Delete all matching TXT records
         for record in resp.result {
-            self.client
-                .request(&DeleteDnsRecord {
-                    zone_identifier: &zone_id,
-                    identifier: &record.id,
-                })
-                .await?;
+            if let DnsContent::TXT { .. } = &record.content {
```

Comment:
Maybe smth like this would be better?

```
if !matches!(&record.content, DnsContent::TXT(_)) {
 continue;
}
```

One less indentation :)

Reply:
let's use even more idiomatic approach then:
```rust
  for record in resp
      .result
      .into_iter()
      .filter(|r| matches!(&r.content, DnsContent::TXT { .. }))
  {
      ...
  }
```